flang/lib/Optimizer/CodeGen/Target.cpp

```diff
@@ -1127,6 +1127,30 @@ struct TargetLoongArch64 : public GenericTarget<TargetLoongArch64> {
     }
     return marshal;
   }
+
+  CodeGenSpecifics::Marshalling
+  integerArgumentType(mlir::Location loc,
+                      mlir::IntegerType argTy) const override {
+    if (argTy.getWidth() == 32) {
+      // LA64 LP64D ABI requires unsigned 32 bit integers to be sign extended.
```

Comment:
The comment and PR talk about "unsigned 32 bit integers", but the check above only checks that this is a "32 bit integer", doesn't the "unsigned" aspect of `argTy` also needs to be checked somehow?

Reply:
Firstly, thanks for your review.
However, I think the check is not necessary. For "signed" argTy, sign extension is natural, and for "unsigned" argTy, sign extension is also required according to LoongArch64 ABI. Therefore, under this ABI, sign extension is always needed for i32 types.